### PR TITLE
update troubleshoot-hot-spot-issues.md

### PR DESCRIPTION
**Pull request by WalterWj**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3276 
- Other reference link(s):

### CODE REVIEW

**Comment by yikeke** [comment]
Hi the original Chinese PR has only two applicable versions - dev and v4.0. But you ticked the 3.1 version in this PR, did you do it on purpose? @WalterWj

File: troubleshoot-hot-spot-issues.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+### Table of hot spots
+
+From TiDB coding rules, The data of the same table will be in a range prefixed by the beginning of the table id, and the order of the data is arranged in the order of the RowID values. When RowID values are incremented during table insert, The inserted line can only be appended at the end. When the region reaches a certain size, it will split, then it can only be appended at the end of the range range, and can always be insert on one region to form a hot spot.
+
+The common increment type self-increment primary key is the sequential increment, by default, when the primary key is integer type, will use the primary key value as the RowID, at this time the RowID is the sequential increment, in a large number of insert to form writing hot spot of the table.
+
+Meanwhile, RowID default in the TiDB is incremented in the order of self-increment. When the primary key is not an integer type, the writing hot spot will also be encountered.
+
+### Index hot spots
```
Comment (lilin90) [inline troubleshoot-hot-spot-issues.md:42]:
```suggestion
### Index hotspots
```

File: troubleshoot-hot-spot-issues.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+Index hot spots are similar to table hot spots, and common hot spots appear in fields that are monotonously increasing in time order, or insert scenes with a large number of repeated values.
+
+## Identification of hot issues
```
Comment (lilin90) [inline troubleshoot-hot-spot-issues.md:46]:
```suggestion
## Identify hotspot issues
```

File: troubleshoot-hot-spot-issues.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+Performance problems are not necessarily caused by hot spots, there may be a number of factors together, before checking need to confirm whether it is related to hot spots.
+
+- The basis for judging and writing hot spots:Open the Hot Write panel in the TiKV-Trouble-Shooting of the monitoring panel (as shown in the figure below) to observe whether there is a phenomenon that the index of individual TiKV nodes is significantly higher than that of other nodes in the monitoring Raftstore CPU.
+
+- Read hot spot basis: open the monitor panel TIKV-Details Thread_CPU, see if there is any obvious tikv particularly high.
+
+### Use TiDB Dashboard to locate the hot tables
```
Comment (lilin90) [inline troubleshoot-hot-spot-issues.md:54]:
```suggestion
### Use TiDB Dashboard to locate hotspot tables
```

File: troubleshoot-hot-spot-issues.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+![Dashboard Example 3](/media/troubleshoot-hot-spot-issues-3.png)
+
+Move the mouse over the bright block, You can see what table or index has a lot of traffic, for example:
+
+![Dashboard Example 4](/media/troubleshoot-hot-spot-issues-4.png)
+
+> [Prior to version 4.0 hot spot location can refer to this document](https://book.tidb.io/session4/chapter7/hotspot-resolved.html)
+
+## Use SHARD_ROW_ID_BITS to process hot spots
```
Comment (lilin90) [inline troubleshoot-hot-spot-issues.md:74]:
```suggestion
## Use `SHARD_ROW_ID_BITS` to process hotspots
```

File: troubleshoot-hot-spot-issues.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+Performance problems are not necessarily caused by hot spots, there may be a number of factors together, before checking need to confirm whether it is related to hot spots.
+
+- The basis for judging and writing hot spots:Open the Hot Write panel in the TiKV-Trouble-Shooting of the monitoring panel (as shown in the figure below) to observe whether there is a phenomenon that the index of individual TiKV nodes is significantly higher than that of other nodes in the monitoring Raftstore CPU.
+
+- Read hot spot basis: open the monitor panel TIKV-Details Thread_CPU, see if there is any obvious tikv particularly high.
+
+### Use TiDB Dashboard to locate the hot tables
+
+The "Hot spot Visualization" feature in TiDB Dashboard helps users narrow down hot spot screening to table level,  Here is an example of the thermal diagram shown by the Hotspot Visualization function, where the horizontal coordinates are time, and the vertical coordinates arrange tables and indexes, The brighter the color, the greater the flow. You can switch to display read or write traffic in the toolbar.
```
Comment (lilin90) [inline troubleshoot-hot-spot-issues.md:56]:
```suggestion
The "Key Visualizer" feature in TiDB Dashboard helps users narrow down hot spot screening to table level,  Here is an example of the thermal diagram shown by the Hotspot Visualization function, where the horizontal coordinates are time, and the vertical coordinates arrange tables and indexes, The brighter the color, the greater the flow. You can switch to display read or write traffic in the toolbar.
```

File: troubleshoot-hot-spot-issues.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+Visible from the flow chart, after setting the SHARD_ROW_ID_BITS, the flow hotspots become very scattered from the previous only on one Region.
+
+## Handle self-increasing primary key hotspots using AUTO_RANDOM
+
+AUTO_RANDOM processing of self-increased primary key hot spot table is suitable to replace self-increased primary key and solve the writing hot spot brought by self-increased primary key.
+
+> The function is currently experimental and is not recommended for use in the production environment. Can be enabled using the following configuration:
+>
+> [experimental]\
+> allow-auto-random = true
```
Comment (lilin90) [inline troubleshoot-hot-spot-issues.md:110]:
```suggestion
> **Note:**
>
> Currently, this is an experimental feature, so it is not recommended to use it in the production environment. To enable this feature, use the following configuration:
>
> ```
> [experimental]\
> allow-auto-random = true
> ```
```

**Comment by lilin90** [comment]
> Hi the original Chinese PR has only two applicable versions - dev and v4.0. But you ticked the 3.1 version in this PR, did you do it on purpose? @WalterWj

I've checked the Chinese documentation and saw that this file does not exist in v3.1, so I unticked v3.1.

File: troubleshoot-hot-spot-issues.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+title: Troubleshoot Hotspot Issues
+summary: Learn how to locate and resolve read or write hotspot issues in TiDB.
+---
+
+# Troubleshoot Hotspot Issues
+
+This document describes how to locate and resolve the problem of read and write hotspots.
+
+As a distributed database, TiDB has a load balancing mechanism to distribute the application loads as evenly as possible to different computing or storage nodes, to make better use of server resources.
+However, in certain scenarios, some application loads cannot be well distributed, which can affect the performance and form a single point of high load, also known as a hotspot.
```
Comment (lilin90) [inline troubleshoot-hot-spot-issues.md:11]:
```suggestion
As a distributed database, TiDB has a load balancing mechanism to distribute the application loads as evenly as possible to different computing or storage nodes, to make better use of server resources. However, in certain scenarios, some application loads cannot be well distributed, which can affect the performance and form a single point of high load, also known as a hotspot.
```

**Pull request merged**